src/tcp_connections.rs

```diff
@@ -76,6 +78,36 @@ pub fn listen(port: u16) -> IoResult<(Receiver<(TcpStream, SocketAddr)>, TcpList
 
     let tcp_listener2 = try!(tcp_listener.try_clone());
     let _ = thread::Builder::new().name("TCP listener".to_string()).spawn(move || {
+        for ifa in getifaddrs::getifaddrs() {
+            let ifa = match ifa.addr {
+                IpAddr::V4(a) => a,
+                _ => continue,
+            };
+            match tcp_listener2.local_addr() {
+                Ok(addr) => {
+                    match addr {
+                        SocketAddr::V4(addr) => {
+                            let port = addr.port();
+                            let _ = thread::Builder::new()
+                                .name("IGD".to_string()).spawn(move || {
+                                match igd::search_gateway_from(ifa) {
+                                    Ok(gateway) => {
+                                        let _ = gateway
+                                            .add_port(igd::PortMappingProtocol::TCP,
+                                                      port,
+                                                      SocketAddrV4::new(ifa, port),
+                                                      0, "crust");
+                                    },
+                                    Err(_) => (),
+                                }
+                            });
+                        },
+                        _ => (),
+                    }
+                },
+                Err(_) => (),
+            }
+        }
         loop {
```

Comment:
Nice work ! @vinipsmaker 
I understand your thoughts regarding the design here, but adding port mapping utility to *transport.rs* makes it protocol agnostic. That means we can have port mapping facility for any new protocol (if supported) automatically in place.